extensions/iam/distributed-identity/identity-did-spi/src/main/java/org/eclipse/dataspaceconnector/iam/did/spi/resolution/DidPublicKeyResolver.java

```diff
@@ -11,9 +10,35 @@
     String FEATURE = "edc:identity:public-key-resolver";
 
     /**
-     * Resolves the public key. Note null is returned if the key cannot be resolved.
+     * Resolves the public key.
      */
-    @Nullable
-    PublicKeyWrapper resolvePublicKey(String did);
+    Result resolvePublicKey(String did);
 
+    /**
+     * The response of a resolve operation.
+     */
+    class Result {
```

Comment:
I think we should standardize the way to return something that could be either a Result or an Error, there's a lot of `*Result` and `*Response`, maybe we could find a common naming convention and, why not, a common abstraction